src/math/power.c

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+ /* Include Files */
+#include <sof/math/power.h>
+
+/* Function Definitions */
+/**
+ * Arguments	: int64_t denominator
+ *
+ * Return Type	: int32_t
+ */
+static inline int32_t div_s32s64(int64_t denominator)
+{
+	uint64_t b_denominator;
+	int32_t quotient;
+
+	if (denominator == 0) {
+		quotient = INT32_MAX;
+	} else {
+		b_denominator =
+		    (denominator < 0) ? ~(uint64_t)denominator + 1 : (uint64_t)denominator;
+		b_denominator = B_NUMERATOR / b_denominator;
+		if (denominator < 0)
+			quotient = (int32_t)-(int64_t)b_denominator;
+		else
+			quotient = (int32_t)b_denominator;
+	}
+	return quotient;
+}
+
+/**
+ * Arguments	: int32_t b
+ * base input range [-32 to +32]
+ *
+ *		: int32_t e
+ * exponent input range [ -3 to +3 ]
+ *
+ * Return Type	: int32_t
+ * power output range [-32768 to + 32768]
+ *
+ * +------------------+------------------+------------------+--------+--------+--------+
+ * | base	      | exponent	 | power(returntype)|	b    |	e     |	 p     |
```

Comment:
im not sure this line is formatted correct, dropped it in sublime text and didn't align properly but might have been my settings

Reply:
I am not sure, if any indentation is missing . Please suggest - I will have a relook into my editor . Thank you